CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+cmake_minimum_required(VERSION 3.10)
+project(spheretree-dist VERSION 1.0 LANGUAGES CXX C)
+
+# Set the required C/C++ standards
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pthread")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+set(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS}")
+set(CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG}")
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-int-in-bool-context -Wsign-compare")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fsanitize=address")
+    set(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS}")
+    set(CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fsanitize=address")
+endif()
+
+
+SET(CMAKE_BUILD_TYPE "Release")
+SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3 -Wall -fPIC -DNDEBUG")
+
+# Include all subdirectories that are part of the project
+add_subdirectory(src/API)
+add_subdirectory(src/Base)
+add_subdirectory(src/BBox)
+add_subdirectory(src/Exceptions)
+add_subdirectory(src/Geometry)
+add_subdirectory(src/LinearAlgebra)
+add_subdirectory(src/MedialAxis)
+add_subdirectory(src/MinMax)
+add_subdirectory(src/SphereTree)
+add_subdirectory(src/Storage)
+add_subdirectory(src/Surface)
+add_subdirectory(src/Export)
+
+# Add the static library equivalent to libSTG.la
+add_library(STG STATIC
+        src/null.c
+)
+
+# Link dependent libraries to libSTG
+target_link_libraries(STG
+        API
+        Base
+        BBox
+        Exceptions
+        Geometry
+        Matrix
+        MedialAxis
+        MinMax
+        SphereTree
+        Storage
+        Surface
+        Export
+)
+
+# Define the list of programs to be built
+set(PROGRAMS
+        makeTreeMedial
+        makeTreeHubbard
+        makeTreeOctree
+        makeTreeGrid
+        makeTreeSpawn
+)
+
+# Add each program and its respective source files
+foreach(PROGRAM ${PROGRAMS})
```

Comment:
Why loop over programs? It would been simpler to do `add_executable` for each program and avoid conditionals.